From 4884383f24ccc109996e695e19dfffabbcd29536 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:37:27 -0400
Subject: [virt] kvm/nvmx: correctly set tr base on nested vmexit emulation

Message-id: <1406040016-3289-53-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86131
O-Subject: [RHEL7 PATCH v2 052/221] KVM: nVMX: correctly set tr base on nested vmexit emulation
Bugzilla: 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Gleb Natapov <gleb@redhat.com>

After commit 21feb4eb64e21f8dc91136b91ee886b978ce6421 tr base is zeroed
during vmexit. Set it to L1's HOST_TR_BASE. This should fix
https://bugzilla.kernel.org/show_bug.cgi?id=60679

Reported-by: Yongjie Ren <yongjie.ren@intel.com>
Reviewed-by: Arthur Chunqi Li <yzt356@gmail.com>
Tested-by: Yongjie Ren <yongjie.ren@intel.com>
Signed-off-by: Gleb Natapov <gleb@redhat.com>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit 205befd9a5c701b56f569434045821f413f08f6d)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kvm/vmx.c b/arch/x86/kvm/vmx.c
index 17caa24..2068d70 100644
--- a/arch/x86/kvm/vmx.c
+++ b/arch/x86/kvm/vmx.c
@@ -8079,7 +8079,7 @@ static void load_vmcs12_host_state(struct kvm_vcpu *vcpu,
  seg.base = vmcs12->host_gs_base;
  vmx_set_segment(vcpu, &seg, VCPU_SREG_GS);
  seg = (struct kvm_segment) {
-  .base = 0,
+  .base = vmcs12->host_tr_base,
   .limit = 0x67,
   .selector = vmcs12->host_tr_selector,
   .type = 11,
-- 
1.7.1